tests/security/selinux/fixfiles.pm

```diff
@@ -17,30 +16,23 @@ sub run {
 
     select_serial_terminal;
 
-    # test `fixfiles check` can print any incorrect file context labels
+    # `fixfiles check` prints any incorrect file context labels
     assert_script_run("fixfiles check > $file_output 2>&1", timeout => 300);
 
-    # pick up a test file to test. We need to use sed+awk to deal with potential spaces in the file path.
-    my $file_info = script_output("grep -i 'Would relabel' $file_output | tail -1 | sed 's/Would relabel //'");
-    my $file_name = script_output("echo $file_info | awk \'{split(\$0, array, \" from \"); print array[1]}\'");
-    my $fcontext_pre = script_output("echo $file_info | awk \'{split(\$0, array, \" from \"); print array[2]}\' | cut -f1 -d' '");
-    my $fcontext_post = script_output("echo $file_info | awk \'{split(\$0, array, \" from \"); print array[2]}\'| cut -f3 -d' '");
+    # pick up a sample file to check the 'restore' feature
+    my $last_line = script_output("grep -i 'Would relabel' $file_output | tail -1");
+    my ($file_name, $fcontext_pre, $fcontext_post) = $last_line =~ m{^Would relabel\s+(.+?)\s+from\s+(\S+)\s+to\s+(\S+)$};
 
     # test `fixfiles restore`: run fixfiles restore on the test file and check the results
-    $self->fixfiles_restore("$file_name", "$fcontext_pre", "$fcontext_post");
+    $self->fixfiles_restore($file_name, $fcontext_pre, $fcontext_post);
 
     # test `fixfiles verify/check`: to double confirm, there should be nothing to do with $file_name
-    my $script_output = script_output("fixfiles verify $file_name", proceed_on_failure => 1);
-    if ($script_output =~ m/$file_name/) {
-        record_info("ERROR", "verify $file_name, it is not well restored: $script_output", result => "fail");
-        $self->result("fail");
-    }
-
-    $script_output = script_output("fixfiles check $file_name", proceed_on_failure => 1);
-    if ($script_output =~ m/$file_name/) {
-        record_info("ERROR", "check $file_name, it is not well restored: $script_output", result => "fail");
-        $self->result("fail");
+    for my $task (qw(verify check)) {
+        my $script_output = script_output("fixfiles $task $file_name", proceed_on_failure => 1);
+        die "$task $file_name, it is not well restored: $script_output" if ($script_output =~ m/$file_name/);
     }
+    # cleanup
+    assert_script_run "rm -f $file_output";
```

Comment:
No need for `-f` here, so better remove it:

```suggestion
    assert_script_run "rm $file_output";
```

Reply:
done at https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/23279